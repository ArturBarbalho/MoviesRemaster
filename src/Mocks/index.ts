import { BannerProps, CardProps, bigbannerProps } from "../typesProps";
export const BiggerBannerMock:bigbannerProps = {
  bigsrc: 'https://image.tmdb.org/t/p/original//9VIURlZvfnmFACs2m0MdUDDC3In.jpg',
  smallsrc:'https://image.tmdb.org/t/p/original//ifRFLx83Xk1DcwAS3OScgI6HmWO.jpg',
  genres:['Action', 'Comedy'],
  id:2,
  media:'movie',
  overview:'Lorem ipsum dolor sit amet, consectetur adipisci elit, sed eiusmod tempor incidunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrum exercitationem ullam corporis...',
  title:'ONYANKOPON 3 O IN√çCIO'
}
export const CardMock:CardProps = {
  id:2,
  media:'movie',
  src:'http://www.claquete.com/fotos/filmes/poster/12861_medio.jpg',
  title:'ONYANKOPON 2 O RETORNO'
}

export const BannerMock:BannerProps = {
    media: 'movie',
    relaseDate: '(1800)',
    id:1,
    title:'ONYANKOPON',
    src:'https://image.tmdb.org/t/p/original//9VIURlZvfnmFACs2m0MdUDDC3In.jpg',
    overview:'Lorem ipsum dolor sit amet, consectetur adipisci elit, sed eiusmod tempor incidunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrum exercitationem ullam corporis...',
    genres: ['Action', 'Comedy' ]
  }