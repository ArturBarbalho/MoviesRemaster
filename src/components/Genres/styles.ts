import styled, { css } from 'styled-components'

type props = {
    size:'banner'|'details'|'bigger'
}

const Modifier = {
    banner:()=>css`
    font-size:larger;
    max-width:250px;
    margin-bottom:30px;
    margin-top:30px;
    display: inline-block;
    margin-left:10px;
    margin-right:10px;
   
    `,
    details:()=>css`
    display: inline-block;
    margin-left:10px;
    margin-bottom:30px;
    `,
    bigger:()=>css`
    margin-left:30px;
    font-size:40px;
    @media(max-width:1400px){       
    font-size:25px;                
    }
    @media(max-width:900px){    
    font-size:20px;
    }
    @media(max-width:750px){
    margin-left:5px;
    }
    `,
}

export const Genre = styled.div<props>`
${({size})=>css`
${ !!size && Modifier[size]()}
color:white;
`}
`

const WrapperModifier = {
    bigger:()=>css`
    display:flex;
    flex-direction:row;
    justify-content:center;
    @media(max-width:750px){
    flex-direction:column;
}
    `,
    banner:()=>css`
    
    `,
    details:()=>css`
    display: inline-block;
    
    `
}

export const Wrapper = styled.div<props>`
${({size})=>css`
${!!size && WrapperModifier[size]()}
`}
`