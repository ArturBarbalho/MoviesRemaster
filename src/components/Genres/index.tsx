import * as S from './styles'
type props = {
genres: string[]
size:'banner'|'details'|'bigger'
}
const Genres = ({genres,size}:props) => (

  <S.Wrapper size={size}>
    {genres.map((e)=>{
    return(
    <S.Genre size={size}
    key={e}>
      {e}
    </S.Genre>
    )})} 

  </S.Wrapper>
)

export default Genres