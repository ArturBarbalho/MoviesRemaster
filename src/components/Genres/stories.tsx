import { Story, Meta } from '@storybook/react/types-6-0'
import Genres from '.'
import { BannerMock } from '../../Mocks'

export default {
  title: 'Genres',
  component: Genres
} as Meta

export const Default: Story = () => <Genres size='details' genres={BannerMock.genres} />