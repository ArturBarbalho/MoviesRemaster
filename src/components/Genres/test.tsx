import { render, screen } from '@testing-library/react'
import Genres from '.'
import { BannerMock } from '../../Mocks'

describe('<Genres />', () => {
  it('should render the right genres', () => {
    const { container } = render(<Genres size='details' genres={BannerMock.genres} />)

    expect(screen.getByText('Action')).toBeInTheDocument()
    expect(screen.getByText('Comedy')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
  it('should render the details size', () => {
    const { container } = render(<Genres size='details' genres={BannerMock.genres} />)

    expect(screen.getByText('Action')).toHaveStyle(
      'margin-left:10px;'
    )
    

    expect(container.firstChild).toMatchSnapshot()
  })
  it('should render the banner size', () => {
    const { container } = render(<Genres size='banner' genres={BannerMock.genres} />)

    expect(screen.getByText('Comedy')).toHaveStyle(
      'max-width:250px;'
    )

    expect(container.firstChild).toMatchSnapshot()
  })
  it('should render the bigger size', () => {
  const { container } = render(<Genres size='bigger' genres={BannerMock.genres} />)

    expect(screen.getByText('Action')).toHaveStyle(
      'font-size:40px;'
    )
    

    expect(container.firstChild).toMatchSnapshot()
  })

})