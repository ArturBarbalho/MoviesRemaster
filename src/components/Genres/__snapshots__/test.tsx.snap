// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Genres /> should render the banner size 1`] = `
<div
  class="styles__Wrapper-sc-6y5ihb-1"
>
  <div
    class="styles__Genre-sc-6y5ihb-0 bOzgdm"
  >
    Action
  </div>
  <div
    class="styles__Genre-sc-6y5ihb-0 bOzgdm"
  >
    Comedy
  </div>
</div>
`;

exports[`<Genres /> should render the bigger size 1`] = `
<div
  class="styles__Wrapper-sc-6y5ihb-1 daGpDx"
>
  <div
    class="styles__Genre-sc-6y5ihb-0 egsKaq"
  >
    Action
  </div>
  <div
    class="styles__Genre-sc-6y5ihb-0 egsKaq"
  >
    Comedy
  </div>
</div>
`;

exports[`<Genres /> should render the details size 1`] = `
<div
  class="styles__Wrapper-sc-6y5ihb-1"
>
  <div
    class="styles__Genre-sc-6y5ihb-0 gCAQid"
  >
    Action
  </div>
  <div
    class="styles__Genre-sc-6y5ihb-0 gCAQid"
  >
    Comedy
  </div>
</div>
`;

exports[`<Genres /> should render the right genres 1`] = `
<div
  class="styles__Wrapper-sc-6y5ihb-1"
>
  <div
    class="styles__Genre-sc-6y5ihb-0 gCAQid"
  >
    Action
  </div>
  <div
    class="styles__Genre-sc-6y5ihb-0 gCAQid"
  >
    Comedy
  </div>
</div>
`;
